pkg/testcase/daemonset.go

```diff
@@ -17,17 +21,77 @@ func TestDaemonset(deployWorkload bool) {
 		Expect(err).NotTo(HaveOccurred(),
 			"Daemonset manifest not deployed")
 	}
-	nodes, _ := shared.ParseNodes(false)
 	pods, _ := shared.ParsePods(false)
 
-	var allNodes []shared.Node
-	for _, node := range nodes {
-		if !(node.Roles == "control-plane,master" || node.Roles == "etcd") {
-			allNodes = append(allNodes, node)
+	cmd := fmt.Sprintf(`
+		kubectl get pods -n test-daemonset -o wide --kubeconfig="%s" \
+		| grep -A10 NODE | awk 'NR>1 {print $7}'
+		`,
+		shared.KubeConfigFile,
+	)
+	nodeNames, err := shared.RunCommandHost(cmd)
+	if err != nil {
+		return
+	}
+
+	var nodes []string
+	n := strings.Split(nodeNames, "\n")
+	for _, nodeName := range n {
+		if nodeName != "" {
+			nodes = append(nodes, nodeName)
```

Comment:
We can probably use `ParsePods` func here, but maybe modify it to accept a namespace. I don't know if it's necessary, but it has some re-usability in that case. I'm fine with this though as it works and is clear what data it is grabbing too 👍

Reply:
parse pods for doing what? sorry

Reply:
That function currently looks like it retrieves the Node name as well, so it could potentially save a bit of logic in this case if that function could take in a namespace and return the struct of pods as it does already. For easy reference: https://github.com/rancher/distros-test-framework/blob/main/shared/cluster.go#L263-L292

Reply:
isnt the same as here  ? 
```

	cmd := fmt.Sprintf(`
		kubectl get pods -n test-daemonset -o wide --kubeconfig="%s" \
		| grep -A10 NODE | awk 'NR>1 {print $7}'
		`,
		shared.KubeConfigFile,
	)
	nodeNames, err := shared.RunCommandHost(cmd)
	if err != nil {
		return
	}
```

Reply:
yeah the end result is still achieved in this case! I don't think you need to change anything if you don't want to here, but just thought it might be helpful for this and potentially future work if we ever need to retrieve similar data on pods, then we could just use ParsePods and pass the desired namespace